lib/facter/java_version.rb

```diff
@@ -11,6 +11,11 @@
 #
 # Notes:
 #   None
+osfamily = Facter.value(:osfamily)
+case osfamily
+when 'OpenBSD'
+  ENV['PATH'] = ENV['PATH'] + File::PATH_SEPARATOR + '/usr/local/jdk-1.7.0/jre/bin'
```

Comment:
Also, since the system may have the `jre` package installed instead of the `jdk` package, it might be a good idea to do this:
```ruby
Facter.add(:java_version) do
  confine :osfamily => 'OpenBSD'
  setcode do
    java_path = Facter::Util::Resolution.which('/usr/local/jdk-1.7.0/jre/bin/java') || Facter::Util::Resolution.which('/usr/local/jre-1.7.0/bin/java')
    if java_path
      Facter::Util::Resolution.exec("#{java_path} -Xmx8m -version 2>&1").lines.first.split(/"/)[1].strip
    end
  end
end
```